zebra-chain/src/transparent.rs

```diff
@@ -149,8 +162,8 @@ pub struct Output {
 }
 
 impl Output {
-    /// Get the value balance from this output
-    pub fn value_balance(&self) -> Amount<NonNegative> {
-        self.value
+    /// Get the value balance of this output
+    pub fn value_balance(&self) -> Amount<NegativeAllowed> {
```

Comment:
Minor: same here, would it make sense to keep the `NonNegative` constraint? I know it might make ergonomics a bit harder, but that might mean it's a good idea to have a `From<Amount<NonNegative>> for Amount<NegativeAllowed>?`

Reply:
> that might mean it's a good idea to have a `From<Amount<NonNegative>> for Amount<NegativeAllowed>?`

There's an existing `Amount::constrain` method. I'd like to keep using it, because is more explicit than `into` or `from`.